systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/PodUtils.java

```diff
@@ -62,32 +62,42 @@ public static void waitForPodsReady(String namespaceName, LabelSelector selector
         waitForPodsReady(namespaceName, selector, expectPods, containers, () -> { });
     }
 
-    public static void waitForPodsReady(String namespaceName, LabelSelector selector, int expectPods, boolean containers, Runnable onTimeout) {
+
+    /**
+     * Waits for a specified number of pods in a given namespace to become ready.
+     *
+     * @param namespace The name of the Kubernetes namespace in which to wait for pods.
+     * @param selector The label selector to identify the pods of interest.
+     * @param expectPods The expected number of pods that should be ready.
+     * @param containers If true, checks individual container readiness within the pods.
+     * @param onTimeout The action to be executed if the pods do not become ready before the timeout.
+     */
+    public static void waitForPodsReady(String namespace, LabelSelector selector, int expectPods, boolean containers, Runnable onTimeout) {
         TestUtils.waitFor("readiness of all Pods matching: " + selector,
             TestConstants.POLL_INTERVAL_FOR_RESOURCE_READINESS, ResourceOperation.timeoutForPodsOperation(expectPods),
             () -> {
-                List<Pod> pods = kubeClient(namespaceName).listPods(namespaceName, selector);
+                List<Pod> pods = kubeClient(namespace).listPods(namespace, selector);
```

Comment:
Why this unnecessary change? In the whole `systemtest` we mostly use `namespaceName` for String based.